### PR TITLE
MR-2566: Rate-supporting documents not converting to Contract-supporting when switching submission types

### PR DESCRIPTION
**Pull request by JasonLin0991**
## Summary
[MR-2566](https://qmacbis.atlassian.net/browse/MR-2566)

Fixed a bug where converting `Rate-supporting` docs to `Contract-supporting` when changing from `Contract and Rate` submission to `Contract Only` was not displaying previously `Rate-supporting` documents as `Contract supporting documents` on the `Review and Submit` page. 

Add code to convert `Rate-supporting` docs to `Contract-supporting` in `updateHealthPlanFormData.ts` resolver and removing code in `FileUpload.tsx` that converted `Rate-supporting` docs to `Contract-supporting` when changing from `Contract and Rate` submission to `Contract Only`. This will handle the converting of documents at the time of changing from `Contract and Rate` submission to `Contract Only` on the `Submission Type` page.

#### Related issues
[2532](https://qmacbis.atlassian.net/browse/MR-2532) This PR will also refactors the code that would remove of `Rate-supporting` only document to instead only convert them to `Contract-supporting` if for some reason the documents had not been converted at the time submission type was changed.

#### Test cases covered

`updateHealthPlanPackage.test.ts`
- `'converts RATES_RELATED documents to CONTRACT_RELATED on contract only submissions'`

### CODE REVIEW

**Review by mojotalantikite — APPROVED** [review]

File: services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts [hunk 1]
```diff
@@ -251,28 +251,18 @@ const generateRateName = (
...
-            if (ratesRelatedDocIndex !== -1) {
-                document.documentCategories.splice(ratesRelatedDocIndex, 1)
-            }
-            noRateDocs.push(document)
-        }
-    })
-    return noRateDocs
+            document.documentCategories.includes('RATES_RELATED')
+                ? ['CONTRACT_RELATED']
+                : ['CONTRACT'],
```
Comment (haworku) [inline services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:264]: What  are we doing with rate certifications now - RATES docs? If I'm reading this right it looked like they are being converted into the CONTRACT type? I think we might want to run this by design if you haven't already

File: services/app-api/src/resolvers/updateHealthPlanFormData.ts [hunk 2]
```diff
@@ -188,6 +191,23 @@ export function updateHealthPlanFormDataResolver(
...
 
+        // If CONTRACT_ONLY submission has supporting documents with RATES_RELATED category we convert this to CONTRACT_RELATED.
+        // This was done on the front end in FileUpload.tsx, but we are doing this here to convert these documents when
+        // the submission type has changed in order to save this across the forms because specifically the review and
+        // submit page, will not display documents that are RATE_RELATED under Contract Details sections when changing
+        // submission type to CONTRACT_ONLY.
+        if (
+            unlockedFormData.submissionType === 'CONTRACT_ONLY' &&
+            unlockedFormData.documents.some((document) =>
+                document.documentCategories.includes('RATES_RELATED')
```
Comment (haworku) [inline services/app-api/src/resolvers/updateHealthPlanFormData.ts:202]: what if document categories include RATES ?

**Review by haworku — COMMENTED** [review]
Things look good otherwise but have questions about the handling of RATES (the rate certification documents).

File: services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts (hunk 1 above)
Comment (JasonLin0991) [reply services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:264]: Good point. We want to remove rate certifications if we are changing submissions types. I thinking to scope this function to only work with supporting documents, 'CONTRACT_RELATED, RATES_RELATED', and throwing errors when certification documents are detected.

Comment (JasonLin0991) [reply services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:264]: Removing rate certifications is done at submit/resubmit in `submitHealthPlanPackage.ts` resolver

**Review by haworku — APPROVED** [review]

File: services/app-api/src/resolvers/updateHealthPlanFormData.ts (hunk 2 above)
Comment (haworku) [reply services/app-api/src/resolvers/updateHealthPlanFormData.ts:202]: What you said is correct. I forgot about what is happening on submit with this workflow.

**Pull request merged**